lib/jekyll/readers/post_reader.rb

```diff
@@ -36,10 +36,15 @@ def read_posts(dir)
     def read_publishable(dir, magic_dir, matcher)
       read_content(dir, magic_dir, matcher).tap { |docs| docs.each(&:read) }
         .select do |doc|
-          site.publisher.publish?(doc).tap do |will_publish|
-            if !will_publish && site.publisher.hidden_in_the_future?(doc)
-              Jekyll.logger.debug "Skipping:", "#{doc.relative_path} has a future date"
+          if doc.content.valid_encoding?
+            site.publisher.publish?(doc).tap do |will_publish|
+              if !will_publish && site.publisher.hidden_in_the_future?(doc)
+                Jekyll.logger.debug "Skipping:", "#{doc.relative_path} has a future date"
+              end
             end
+          else
+            Jekyll.logger.debug "Skipping:", "#{doc.relative_path} is no valid UTF-8"
```

Comment:
I wasn't sure if this should rather be a warning? But given that we already have an error message when this file is read i though it's good enough.

Reply:
From a user perspective, a comprehensive warning would be easier to grasp:

```sh
Warning;  Found a binary file in your '_posts' directory, skipping: /path/to/image.jpg
```

Reply:
Why is it a warning? Is something bad happening?

Reply:
I guess @DirtyF changed his suggestion in the meantime to an `info` message? I like it, makes it easier to detect then a `debug` message and the existing `error` message does not include the information that a file has been skipped.